builtin/providers/vsphere/storage_pod_data_store.go

```diff
@@ -0,0 +1,258 @@
+package vsphere
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+	"golang.org/x/net/context"
+)
+
+// Collector models the PropertyCollector managed object.
+//
+// For more information, see:
+// http://pubs.vmware.com/vsphere-55/index.jsp#com.vmware.wssdk.apiref.doc/vmodl.query.PropertyCollector.html
+//
+type StoragePodDataStore struct {
+	name           string
+	template       string
+	storagePodName string
+	clone          bool
+
+	ConfigSpecsNetwork []types.BaseVirtualDeviceConfigSpec
+	ResourcePool       *object.ResourcePool
+	// TODO Do we need this?
+	// HostSystem         *object.HostSystem
+	Folder         *object.Folder
+	VirtualMachine *object.VirtualMachine
+	DataCenter     *object.Datacenter
+	StoragePod     *object.StoragePod
+}
+
+// Based of of govc clone.go
+// Get the recommended StoragePod datastore
+func (spds *StoragePodDataStore) findRecommendedStoragePodDataStore(client *vim25.Client) (datastore *object.Datastore, err error) {
+
+	folderref := spds.Folder.Reference()
+	poolref := spds.ResourcePool.Reference()
+
+	relocateSpec := types.VirtualMachineRelocateSpec{
+		DeviceChange: spds.ConfigSpecsNetwork,
+		Folder:       &folderref,
```

Comment:
where is the Folder parameter coming from, it is not defined in the VirtualMachineRelocateSpec:
https://www.vmware.com/support/developer/vc-sdk/visdk2xpubs/ReferenceGuide/vim.vm.RelocateSpec.html